client/src/Component/SMSModal.js

```diff
@@ -1,23 +1,54 @@
-/* eslint-disable jsx-a11y/label-has-for */
+
 import { useState } from "react";
 import Modal from "react-bootstrap/Modal";
 import Button from "react-bootstrap/Button";
 
+
+
+
 function SMSModal(props) {
     const [show, setShow] = useState(false);
     const hideModal = () => setShow(false);
     const showModal = () => setShow(true);
 
+    let localMessage = localStorage.getItem("message");
+    if (localMessage === null) {
+      localMessage = "Hi, this is Bikes For Refugees. Bikes are now available for you to pick up. Please click LINK to choose a pick up date. For any questions, please contact 07900000000.";
+    }
+    const [message, setMessage] = useState(localMessage);
```

Comment:
```suggestion
    const [message, setMessage] = useState(() => {
      const localMessage = localStorage.getItem("message");
      if (localMessage) {
        return localMessage;
      }
      return defaultMessage;
    });
```
Looking at a [localStorage in React tutorial](https://blog.logrocket.com/using-localstorage-react-hooks/), it looks like if we have multiple statements we want to execute to initialise state, we can include it in a function that we pass to `useState`.

We can also set the default message in `defaultMessage` and declare it outside the `SMSModal` function since we don't need to repeatedly create it on every render.

Reply:
I'm struggling with this bit - I've tried various ways of setting the textarea value and I either come up with something that isn't editable, or it logs the message as typed to the console but doesn't show the change dynamically as it's typed?